Do not build the sample teststore program.

It uses implicit function declarations (fork, sleep, xenstore_*) that
GCC 15 treats as errors, and it is not installed.

Upstream-Status: Inappropriate [OpenXT packaging]

--- a/Makefile.am
+++ b/Makefile.am
@@ -17,8 +17,7 @@
 #

 SUBDIRS = \
-  src     \
-  sample
+  src

 EXTRA_DIST = version-major version-minor version-micro version-files version-md5sums
 bin_SCRIPTS = libxcxenstore-config
